account_invoice_validate_send_email/models/account_invoice.py

```diff
@@ -125,10 +125,10 @@ def _compute_carrier_info(self):
         for invoice in self:
             carrier_recs = invoice.picking_ids.mapped("carrier_info_id")
             if carrier_recs:
-                invoice.carrier_info_name = ", ".join(x.name for x in carrier_recs)
-                invoice.carrier_tracking_url = ", ".join(
-                    x.tracking_url for x in carrier_recs
-                )
+                for x in carrier_recs:
+                    invoice.carrier_info_name = ", ".join(x.name)
+                    if x.tracking_url:
+                        invoice.carrier_tracking_url = ", ".join(x.tracking_url)
```

Comment:
```suggestion
                invoice.carrier_info_name = ", ".join(x.name for x in carrier_recs)
                tracking_urls = carrier_recs.mapped("tracking_url")
                if tracking_urls:
                    invoice.carrier_tracking_url = ", ".join(
                        x.tracking_url for x in tracking_urls
                    )
```

@TomohiroAbe-Quartile Can you check if above makes sense.  Haven't tested it myself.

Reply:
@yostashiro 
I checked and it works as expected.
Situation like below:
1 invoice record has 2 deliveries.
1 delivery has carrier_name and carrier_number.
And other one has carrier_name, but not has carrier_number.

And succeed to validate, and send e-mail.